drafts/2020-02-20-Rest-client-generator.adoc

```diff
@@ -0,0 +1,107 @@
+
+The pet store service documents the API using the OpenAPI specification because it is both human readable and machine readable. You download the API documentation in a file called petstore.yaml. The challenge now is to design and code the Java module that will communicate with the pet store service.
+
+The OpenAPI Generator can write some of that code for you. It can interpret the description in the yaml file and pull out the data model and the API. It will create Java files that describe the data model objects and method interfaces for each of the pet store endpoints.
+
+== The OpenAPI Description
+As an example the following snippet shows an endpoint to add a new pet in the petstore.yaml:
+-----
+/pet:
+  post:
+    summary: Add a new pet to the store
+    consumes:
+      - application/json
+    produces:
+      - application/json
+    parameters:
+      - in: body
+        name: body
+        schema:
+          $ref: '#/definitions/Pet'
+-----
+This endpoint's URL could look something like https://<server>/pet. The http verb is POST, the request's body is JSON and the body conforms to the schema referenced in the 'definitions' portion of the API document:
+-----
+  Pet:
+    description: A pet for sale in the pet store
+    type: object
+    properties:
+      id:
+        type: integer
+      name:
+        type: string
+        example: Rex the dog
+      photoUrls:
+        type: array
+        items:
+          type: string
+-----
+This schema defines three fields: one integer, one string and one array. A sample Pet object (in yaml format) could be:
+-----
+Pet:
+  id: 1234
+  name: Rex
+  photoUrls:
+    - https://photos.com/abcd1234
+-----
+== See Generated Code
+The generator will create model objects in Java for each of the types defined in the petstore.yaml. You can use these objects to transfer data to and from the service. For example, the Java Pet object will be, in part:
+-----
+public class Pet  {
+  private Long id;
+  private String name;
+  private List<String> photoUrls = new ArrayList<String>();
+  ...
+-----
+The generator will also create a Java interface for each endpoint. You can use these methods in your Microprofile front end to access the service. For example, to create a new pet you would call this method:
+-----
+    @POST
+    @Path("/pet")
+    @Consumes({ "application/json" })
+    public void addPet(Pet pet) throws ApiException, ProcessingException;
+-----
+While the OpenAPI Generator will create the framework, you still must implement the methods that will access the service. However, this task is much easier with the design laid out and some of the code written.
+
+== From the Command Line
+
+There are two ways to get started. The first is to download the official build and invoke the generator from the command line. See https://openapi-generator.tech/docs/installation#jar[Download Instructions].
```

Comment:
Is this really the approach we recommend rather than using MicroProfile Rest Client, as described in the guide linked to earlier?

Reply:
The generator technology is actually complementary. The guide says "Create a RESTful client interface for the system service. Write a template interface that maps the API of the remote system service" and this generator creates the interface file. You can run the generator on the command line or in VSCode.

Reply:
Okay, I think this is the explanation that was missing from the opening paragraphs at the top of the post.

Reply:
The opening paragraph explains: "you can use the OpenAPI Generator to generate the REST client code for you"